Sources/SourceControl/GitRepository.swift

```diff
@@ -387,6 +398,33 @@ public class GitRepository: Repository, WorkingCheckout {
         return localFileSystem.isDirectory(AbsolutePath(firstLine))
     }
 
+    /// Returns true if the file at `path` is ignored by `git`
+    public func isGitIgnored(_ paths: [AbsolutePath]) throws -> [Bool] {
+        return try queue.sync {
+            let stringPaths = paths.map({ $0.asString })
+            let pathsFileContent = stringPaths.joined(separator: "\0")
+
+            let pathsFile = try TemporaryFile()
+            let pathsData = Data(pathsFileContent.utf8)
+            pathsFile.fileHandle.write(pathsData)
```

Comment:
We should use BufferedOutputByteStream and localFileSystem to APIs to write to the temp file.